### PR TITLE
driver/mem:add Mem Driver.

### PR DESCRIPTION
**Pull request by yangguangcai01**
## Summary
The `devmem` driver provides an interface for accessing memory-mapped
I/O in an embedded system.
## Impact
N/A
## Testing
ci

### CODE REVIEW

File: drivers/misc/dev_mem.c [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+/****************************************************************************
+ * Name: devmem_read
+ ****************************************************************************/
+
+static ssize_t devmem_read(FAR struct file *filep, FAR char *buffer,
+                           size_t buflen)
+{
+  FAR struct memory_region_s *region = filep->f_inode->i_private;
+  uintptr_t src = filep->f_pos;
```
Comment (acassis) [inline drivers/misc/dev_mem.c:88]: Please include some debug assertions to confirm there is no NULL pointers

File: drivers/misc/dev_mem.c [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+}
+
+/****************************************************************************
+ * Name: devmem_write
+ ****************************************************************************/
+
+static ssize_t devmem_write(FAR struct file *filep, FAR const char *buffer,
+                            size_t buflen)
+{
+  FAR struct memory_region_s *region = filep->f_inode->i_private;
```
Comment (acassis) [inline drivers/misc/dev_mem.c:121]: Ditto

File: drivers/misc/dev_mem.c [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+}
+
+/****************************************************************************
+ * Name: devmem_mmap
+ ****************************************************************************/
+
+static int devmem_mmap(FAR struct file *filep,
+                       FAR struct mm_map_entry_s *map)
+{
+  FAR struct memory_region_s *region = filep->f_inode->i_private;
```
Comment (acassis) [inline drivers/misc/dev_mem.c:155]: Ditto

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by yangguangcai01** [comment]
> @yangguangcai01 please include a commit message about this driver. Also include a Documentation/ to this new driver



> @yangguangcai01 please include a commit message about this driver. Also include a Documentation/ to this new driver

done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**